From b64149c3fc292c1763c3f0056d6a8081e2ce0810 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Mon, 10 Apr 2017 11:17:51 +0200
Subject: [PATCH] sftp: Handle SecurID password prompt

SFTP backend parses output from SSH client and expects several predefined
password prompts. Unfortunately, SecurID (pam_securid.so) is not handled
yet and the mount operation timeouts with the following error: "Timed out
when logging in". Let's handle also SecurID prompt (i.e. "Enter PASSCODE").

https://bugzilla.redhat.com/show_bug.cgi?id=1440256
---
 daemon/gvfsbackendsftp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/daemon/gvfsbackendsftp.c b/daemon/gvfsbackendsftp.c
index 353f092e..22ad4db7 100644
--- a/daemon/gvfsbackendsftp.c
+++ b/daemon/gvfsbackendsftp.c
@@ -1100,7 +1100,8 @@ handle_login (GVfsBackend *backend,
           g_str_has_suffix (buffer, "Password:")  ||
           g_str_has_prefix (buffer, "Password for ") ||
           g_str_has_prefix (buffer, "Enter Kerberos password") ||
-          g_str_has_prefix (buffer, "Enter passphrase for key"))
+          g_str_has_prefix (buffer, "Enter passphrase for key") ||
+          g_str_has_prefix (buffer, "Enter PASSCODE"))
         {
           gboolean aborted = FALSE;
           gsize bytes_written;
-- 
2.12.2

